internal/tailer/tail.go

```diff
@@ -138,19 +153,23 @@ func New(ctx context.Context, wg *sync.WaitGroup, lines chan<- *logline.LogLine,
 		lines:        lines,
 		initDone:     make(chan struct{}),
 		globPatterns: make(map[string]struct{}),
+		logSockets:   make(map[string]struct{}),
 		logstreams:   make(map[string]logstream.LogStream),
 	}
 	defer close(t.initDone)
 	if err := t.SetOption(options...); err != nil {
 		return nil, err
 	}
-	if len(t.globPatterns) == 0 {
-		glog.Info("No patterns to tail, tailer done.")
+	if len(t.globPatterns) == 0 && len(t.logSockets) == 0 {
+		glog.Info("No patterns or sockets to tail, tailer done.")
 		close(t.lines)
 		return t, nil
 	}
 	// Guarantee all existing logs get tailed before we leave.  Also necessary
 	// in case oneshot mode is active, the logs get read!
+	if err := t.PollLogSockets(); err != nil {
```

Comment:
Ah, I never thought that it would connect to existing sockets, and only that it would create its own.  I didn't think a wildcard would be valid for a socket, but it certainly is.

Reply:
Ah yeah.  I've confused myself again.  The unix socket listener has to be open first I believe, so there won't be wildcards.  But I still think the rest of the comments are valid so far as keeping the code as close to `logstream.New` as possible.